dm/pkg/checker/privilege_test.go

```diff
@@ -27,161 +28,272 @@ var _ = tc.Suite(&testCheckSuite{})
 			},
-			dumpState:        StateSuccess,
-			replicationState: StateSuccess,
+			dumpState: StateSuccess,
 		},
 		{
 			grants: []string{ // Aurora have `LOAD FROM S3, SELECT INTO S3, INVOKE LAMBDA`
 				"GRANT SELECT, INSERT, UPDATE, DELETE, CREATE, DROP, RELOAD, PROCESS, REFERENCES, INDEX, ALTER, SHOW DATABASES, CREATE TEMPORARY TABLES, LOCK TABLES, EXECUTE, REPLICATION SLAVE, REPLICATION CLIENT, CREATE VIEW, SHOW VIEW, CREATE ROUTINE, ALTER ROUTINE, CREATE USER, EVENT, TRIGGER, LOAD FROM S3, SELECT INTO S3, INVOKE LAMBDA, INVOKE SAGEMAKER, INVOKE COMPREHEND ON *.* TO 'root'@'%' WITH GRANT OPTION",
 			},
-			dumpState:        StateSuccess,
-			replicationState: StateSuccess,
+			dumpState: StateSuccess,
 		},
 		{
 			grants: []string{ // Aurora have `LOAD FROM S3, SELECT INTO S3, INVOKE LAMBDA`
 				"GRANT INSERT, UPDATE, DELETE, CREATE, DROP, PROCESS, REFERENCES, INDEX, ALTER, SHOW DATABASES, CREATE TEMPORARY TABLES, LOCK TABLES, EXECUTE, CREATE VIEW, SHOW VIEW, CREATE ROUTINE, ALTER ROUTINE, CREATE USER, EVENT, TRIGGER, LOAD FROM S3, SELECT INTO S3, INVOKE LAMBDA, INVOKE SAGEMAKER, INVOKE COMPREHEND ON *.* TO 'root'@'%' WITH GRANT OPTION",
 			},
-			dumpState:        StateFailure,
-			replicationState: StateFailure,
+			dumpState: StateFailure,
+			errMatch:  "lack of Select privilege: {`INFORMATION_SCHEMA`};",
 		},
 		{
 			grants: []string{ // test `LOAD FROM S3, SELECT INTO S3` not at end
 				"GRANT INSERT, UPDATE, DELETE, CREATE, DROP, RELOAD, PROCESS, REFERENCES, INDEX, ALTER, SHOW DATABASES, CREATE TEMPORARY TABLES, LOCK TABLES, EXECUTE, REPLICATION SLAVE, REPLICATION CLIENT, CREATE VIEW, SHOW VIEW, CREATE ROUTINE, ALTER ROUTINE, CREATE USER, EVENT, TRIGGER, LOAD FROM S3, SELECT INTO S3, SELECT ON *.* TO 'root'@'%' WITH GRANT OPTION",
 			},
-			dumpState:        StateSuccess,
-			replicationState: StateSuccess,
+			dumpState: StateSuccess,
 		},
 		{
 			grants: []string{ // ... and `LOAD FROM S3` at beginning, as well as not adjacent with `SELECT INTO S3`
 				"GRANT LOAD FROM S3, INSERT, UPDATE, DELETE, CREATE, DROP, RELOAD, PROCESS, REFERENCES, INDEX, ALTER, SHOW DATABASES, CREATE TEMPORARY TABLES, LOCK TABLES, EXECUTE, REPLICATION SLAVE, REPLICATION CLIENT, CREATE VIEW, SHOW VIEW, CREATE ROUTINE, ALTER ROUTINE, CREATE USER, EVENT, TRIGGER, SELECT INTO S3, SELECT ON *.* TO 'root'@'%' WITH GRANT OPTION",
 			},
-			dumpState:        StateSuccess,
-			replicationState: StateSuccess,
+			dumpState: StateSuccess,
+		},
+		{
+			grants: []string{ // lack db/table level privilege
+				"GRANT ALL PRIVILEGES ON `medz`.* TO `zhangsan`@`10.8.1.9` WITH GRANT OPTION",
+			},
+			dumpState: StateFailure,
+			checkTables: map[string][]string{
+				"medz": {"medz"},
+			},
+			errMatch: "lack of Select privilege: {`INFORMATION_SCHEMA`};",
 		},
 		{
 			grants: []string{ // privilege on db/table level is not enough to execute SHOW MASTER STATUS
```

Comment:
this test is to check that "if a global level privilege is needed, grant it on db/table level is not enough". seems after your change no test will cover this case. Please keep a case for it.

Another reason I require the case is that we maintain the `lackPriv` with this property: "If a privilege is needed to be global granted, it should not have any db/table level entry", and the deletion of db/table level entry of `lackPriv` in `verifyPrivileges` relies on this property. 
That is to say, if in future we mistakenly append a db/table level entry under a need-global-granted privilege to `lackPriv`, `verifyPrivileges` will delete whole need-global-granted privilege by cleaning up zero-length map.

Reply:
This is because we don't test `RELOAD` in UT. I added it.